### PR TITLE
Issue/none set connection runtime parameters for uppercase set

### PR DESCRIPTION
**Pull request by wyrzyk**

### CODE REVIEW

**Review by dwatl — APPROVED** [review]

File: src/main/java/com/atlassian/db/replica/internal/SqlQuery.java [hunk 1]
```diff
@@ -38,9 +39,17 @@ public boolean isInsert() {
...
     public boolean isSqlSet() {
+        return compatibleWithPreviousVersion ? isSqlSet_old() : isSqlSet_new();
+    }
+
+    private boolean isSqlSet_old() {
         return sql.startsWith("set");
     }
 
+    private boolean isSqlSet_new() {
+        return sql.startsWith("set") || sql.startsWith("SET");
```
Comment (afaruga-atlassian) [inline src/main/java/com/atlassian/db/replica/internal/SqlQuery.java:50]: how about `sEt` or `SeT` :lol

File: src/test/java/com/atlassian/db/replica/api/TestDualConnection.java [hunk 2]
```diff
@@ -1658,6 +1658,43 @@ public void shouldKeepUsingReplicaAfterSettingTimeout() throws SQLException {
...
+    @Test
+    public void shouldKeepUsingReplicaAfterSettingTimeoutUpperCase() throws SQLException {
+        final ConnectionProviderMock connectionProvider = new ConnectionProviderMock();
+        final Connection connection = DualConnection.builder(
+            connectionProvider,
+            permanentConsistency().build()
+        ).build();
+
+        try (Statement statement = connection.createStatement()) {
+            statement.execute("SET statement_timeout to 30000");
```
Comment (afaruga-atlassian) [inline src/test/java/com/atlassian/db/replica/api/TestDualConnection.java:1670]: If this is connected "statment timeouts leaks` then... 🤦 🤦

**Review by afaruga-atlassian — APPROVED** [review]

File: src/main/java/com/atlassian/db/replica/internal/SqlQuery.java (hunk 1 above)
Comment (wyrzyk) [reply src/main/java/com/atlassian/db/replica/internal/SqlQuery.java:50]: We had the discussion before 😉. Db-replica, in the first place, aims for low overhead, and correctness is in second place. It will not work correctly for `SeT`, but it's not worth reducing the performance of every query to cover this corner case.

File: src/test/java/com/atlassian/db/replica/api/TestDualConnection.java (hunk 2 above)
Comment (wyrzyk) [reply src/test/java/com/atlassian/db/replica/api/TestDualConnection.java:1670]: It might be connected to the leaks if we used an upper case syntax. The other way it may be incorrect are different ways to call a query (executeUpdate, executeQuery etc.). The test case covers only `Statement#execute`.

File: src/main/java/com/atlassian/db/replica/internal/SqlQuery.java (hunk 1 above)
Comment (afaruga-atlassian) [reply src/main/java/com/atlassian/db/replica/internal/SqlQuery.java:50]: > We had the discussion before
Yeah, we had, bad luck that we didn't find SET that time.
